scripts/mac-signing.sh

```diff
@@ -4,69 +4,76 @@ sudo security import MacCertificate.p12 -k /Library/Keychains/System.keychain -P
+  codesign -s "Developer ID Application: InfluxData Inc. (M7DN9H35QT)" --timestamp --options=runtime ../scripts/telegraf_entry_mac
+  codesign -v ../scripts/telegraf_entry_mac
 
-# Send the DMG to be notarized.
-uuid=$(xcrun altool --notarize-app --primary-bundle-id "com.influxdata.telegraf" --username "$AppleUsername" --password "$ApplePassword" --file "$baseName".dmg | awk '/RequestUUID/ { print $NF; }')
-echo $uuid
-if [[ $uuid == "" ]]; then 
-  echo "Could not upload for notarization."
-  exit 1
-fi
+  # Create the .app bundle.
+  mkdir Telegraf
+  cd Telegraf
+  mkdir Contents
+  cd Contents
+  mkdir MacOS
+  mkdir Resources
+  cd ../..
+  cp ../info.plist Telegraf/Contents
+  cp -R "$extractedFolder"/ Telegraf/Contents/Resources
+  cp ../scripts/telegraf_entry_mac Telegraf/Contents/MacOS
+  cp ../assets/icon.icns Telegraf/Contents/Resources
+  chmod +x Telegraf/Contents/MacOS/telegraf_entry_mac
+  mv Telegraf Telegraf.app
 
-# Wait until the status returns something other than 'in progress'.
-request_status="in progress"
-while [[ "$request_status" == "in progress" ]]; do
-  sleep 10
-  request_status=$(xcrun altool --notarization-info $uuid --username "$AppleUsername" --password "$ApplePassword" 2>&1 | awk -F ': ' '/Status:/ { print $2; }' )
-done
+  # Sign the entire .app bundle, and wrap it in a DMG.
+  codesign -s "Developer ID Application: InfluxData Inc. (M7DN9H35QT)" --timestamp --options=runtime --deep --force Telegraf.app
+  hdiutil create -size 500m -volname Telegraf -srcfolder Telegraf.app "$baseName".dmg
+  codesign -s "Developer ID Application: InfluxData Inc. (M7DN9H35QT)" --timestamp --options=runtime "$baseName".dmg
 
-if [[ $request_status != "success" ]]; then
-  echo "Failed to notarize."
-  exit 1
-fi
+  # Send the DMG to be notarized.
+  uuid=$(xcrun altool --notarize-app --primary-bundle-id "com.influxdata.telegraf" --username "$AppleUsername" --password "$ApplePassword" --file "$baseName".dmg | awk '/RequestUUID/ { print $NF; }')
+  echo $uuid
+  if [[ $uuid == "" ]]; then 
+    echo "Could not upload for notarization."
+    exit 1
+  fi
 
-# Attach the notarization to the DMG.
-xcrun stapler staple "$baseName".dmg
-rm -rf Telegraf.app
-rm -rf $extractedFolder
-ls
+  # Wait until the status returns something other than 'in progress'.
+  request_status="in progress"
+  while [[ "$request_status" == "in progress" ]]; do
+    sleep 10
+    request_status=$(xcrun altool --notarization-info $uuid --username "$AppleUsername" --password "$ApplePassword" 2>&1 | awk -F ': ' '/Status:/ { print $2; }' )
+  done
 
-echo "Signed and notarized!"
+  if [[ $request_status != "success" ]]; then
+    echo "Failed to notarize."
+    exit 1
+  fi
+
+  # Attach the notarization to the DMG.
+  xcrun stapler staple "$baseName".dmg
+  rm -rf Telegraf.app
+  rm -rf $extractedFolder
+  ls
+
+  echo "${macFiles[$tarFile]} Signed and notarized!"
```

Comment:
Shouldn't this be `$tarFile`?

Reply:
I think it would do the same thing but yours is more straightforward.